app/src/core/profile/ProfilesController.ts

```diff
@@ -22,28 +23,27 @@ export default class ProfilesController extends EventEmitter {
 
     // Получение объекта пользователя
     public getProfile(userId: string = MY_ID): Profile {
-        if (!this._profiles.has(userId)) {
-            throw new Error(`Профиля с id: ${userId} не существует.`);
+        if (this._profiles.has(userId)) {
+            return this._profiles.get(userId)!;
         }
 
-        return this._profiles.get(userId) as Profile;
+        this.emit(MainClientEvents.ERROR, `Профиля с id: ${userId} не существует.`);
+        return new EmptyProfile("", this._request, this._dispatch);
```

Comment:
So as I can understand here now we get not only error message but also emty profile page?

Reply:
no, empty profile instance (it is not a page!!!!) we should use to handle exceptional cases.
So in this case we emit the error to the MainClient class (this is the only place to handle the error) and then we should return the profile instance. Because if this method in ProfilesController does not return a Profile instance - we get an error in components which use the useProfile hook